scylla/src/transport/load_balancing/dc_aware_round_robin.rs

```diff
@@ -80,8 +82,11 @@ impl LoadBalancingPolicy for DcAwareRoundRobinPolicy {
             "DC Aware"
         );
 
-        let plan = rotated_local_nodes.chain(rotated_remote_nodes);
-        Box::new(plan)
+        if self.include_remote_nodes {
+            Box::new(rotated_local_nodes.chain(rotated_remote_nodes))
```

Comment:
Let's put the calculation of remote nodes inside this branch so that they do not need to be calculated when not used. The `trace!` command will have to be moved into this branch and the sibling branch, and then modified appropriately.